File: munch-api/app/models/recipe.py
from app import db

class Recipe(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    title = db.Column(db.String(100), nullable=False)
    instructions = db.Column(db.Text, nullable=False)
    author = db.Column(db.String(64), nullable=False)
    imageUrl = db.Column(db.String(255), nullable=False)
    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)

    def __repr__(self):
        return f'<Recipe {self.id}, {self.title}>'
    
    def to_dict(self):
        return {
            'id': self.id,
            'title': self.title,
            'instructions': self.instructions,
            'author': self.author,
            'imageUrl': self.imageUrl,
            'user_id': self.user_id
        }

File: munch-app/src/types/recipe.ts
export interface Recipe {
    id: number;
    title: string;
    instructions: string;
    author: string;
    image_url: string;
}

export interface RecipeData {
    id: number;
    title: string;
    instructions: string;
    author: string;
    image_url: string;
}

export interface RecipeUpdateData {
    title?: string;
    instructions?: string;
    author?: string;
    image_url?: string;
}

export interface IngredientsRequest {
    ingredients: string[];
} 